src/tools/compiletest/src/runtest.rs

```diff
@@ -2749,7 +2749,7 @@ impl<'test> TestCx<'test> {
         // Wrapper tools set by `runner` might provide extra output on failure,
         // for example a WebAssembly runtime might print the stack trace of an
         // `unreachable` instruction by default.
-        let compare_output_by_lines = self.config.runner.is_some();
+        let compare_output_by_lines = self.props.parallel_front_end || self.config.runner.is_some();
```

Comment:
Remark: note that this has two potential issues:

1. This can mask genuine bugs, where a symptom may be diagnostics reordering.
2. The parallel front-end tests are still potential flaky.

I don't love (2) too much, but I recognize the pragmatic need for having some test coverage, so I don't want to block this.

My reservation is primarily that these tests can potentially fail someone else's PR, or a rollup, and can be rather confusing.